src/git/logging_shell.go

```diff
@@ -0,0 +1,115 @@
+package git
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"runtime"
+	"strings"
+
+	"github.com/fatih/color"
+	"github.com/git-town/git-town/src/command"
+	"github.com/git-town/git-town/src/dryrun"
+	"github.com/kballard/go-shellquote"
+)
+
+// LoggingShell is an implementation of the Shell interface
+// that runs commands in the current working directory
+// and streams the command output to the application output.
+// It is used by Git Town commands to run Git commands that show up in their output.
+type LoggingShell struct {
+	currentBranchTracker *CurrentBranchTracker
+}
+
+// NewLoggingShell provides StreamingShell instances.
+func NewLoggingShell(branchTracker *CurrentBranchTracker) *LoggingShell {
+	return &LoggingShell{branchTracker}
+}
+
+// WorkingDir provides the directory that this Shell operates in.
+func (shell LoggingShell) WorkingDir() string {
+	return "."
+}
+
+// MustRun runs the given command and returns the result. Panics on error.
+func (shell LoggingShell) MustRun(cmd string, args ...string) *command.Result {
+	return command.MustRun(cmd, args...)
+}
+
+// Run runs the given command in this ShellRunner's directory.
+func (shell LoggingShell) Run(cmd string, args ...string) (*command.Result, error) {
+	shell.PrintCommand(cmd, args...)
+	if dryrun.IsActive() {
+		if len(args) == 2 && cmd == "git" && args[0] == "checkout" {
+			dryrun.SetCurrentBranchName(args[1])
```

Comment:
In a future PR, I want to move the functionality inside the `dryrun` package into this class and get rid of that package altogether.